Mathlib/Topology/MetricSpace/ProperSpace.lean

```diff
@@ -123,6 +123,13 @@ instance pi_properSpace {π : β → Type*} [Fintype β] [∀ b, PseudoMetricSpa
   rw [closedBall_pi _ hr]
   exact isCompact_univ_pi fun _ => isCompact_closedBall _ _
 
+lemma ProperSpace.of_isClosed {X : Type*} [PseudoMetricSpace X] [ProperSpace X]
+    {s : Set X} (hs : IsClosed s) :
+    ProperSpace s :=
+  ⟨fun x r ↦ Topology.IsEmbedding.subtypeVal.isCompact_iff.mpr
+    ((isCompact_closedBall x.1 r).of_isClosed_subset
+    (hs.isClosedMap_subtype_val _ isClosed_ball) (Set.image_subset_iff.mpr subset_rfl))⟩
```

Comment:
I think I would prove this using `LipschitzWith.properSpace`, by first specializing it to closed isometric maps (which are automatically lipschitz and proper, and I think this intermediate statement is convenient for discoverability), and then apply it to the case of actual subtype inclusions. Is that too much of a mess for the imports?

Reply:
Yes, this is precisely what happened. I needed the result for closed subspaces, generalized the proof to proper lipschitz maps, proceeded to use the general lemma to prove it but then discovered that it will need to live in the lipschiz file and import isometry so I gave up.

Reply:
I added a comment to point to the general lemma.